crates/rustc_codegen_spirv/Cargo.toml

```diff
@@ -30,13 +30,6 @@ use-compiled-tools = ["spirv-tools/use-compiled-tools"]
 # Enable this feature to be able to experiment with other versions.
 skip-toolchain-check = []
 
-# HACK(eddyb) these are not real "build dependencies", but rather dependencies
-# of proc macros, which we want to enable extra features in (without having
-# a proc macro crate to do that through), for the same reasons as the list at
-# the start of `[dependencies]` (avoiding unnecessary rebuilds).
-[build-dependencies]
-syn = { version = "1", features = ["extra-traits", "full"] }
-
```

Comment:
This made absolutely no difference in my testing with syn 2, A being with and B without this dependency:
```
    native  spirv   real    user
A1  43.43   9.35    1m7     9m19    
A2  41.16   8.79    1m5     8m59    
B1  42.48   8.84    1m6     9m1     
B2  42.37   9.00    1m6     9m0    
```

Reply:
There are other similar dependencies below I have not touched.

Reply:
Will discuss this more in-depth off this PR, but for historical context, `cargo clean` with a single Cargo build after it is probably completely unimpacted by what this was fixing.

The issue was switching between things like the `wgpu` example and `compiletest` etc. - and the debugging process to track down the rebuild causes was honestly a bit absurd (there might be better tools now, or Cargo functionality, but I don't know off the top of my head).

Reply:
Example of the issue at play (after running each of these commands once):
```console
$ cargo build -p compiletests --release --no-default-features --features use-installed-tools
   Compiling rustc_codegen_spirv v0.9.0 (/home/eddy/Projects/rust-gpu/crates/rustc_codegen_spirv)
   Compiling compiletests v0.0.0 (/home/eddy/Projects/rust-gpu/tests)
    Finished `release` profile [optimized] target(s) in 2.92s
$ cargo build -p example-runner-wgpu --release --no-default-features --features use-installed-tools
   Compiling rustc_codegen_spirv v0.9.0 (/home/eddy/Projects/rust-gpu/crates/rustc_codegen_spirv)
   Compiling spirv-builder v0.9.0 (/home/eddy/Projects/rust-gpu/crates/spirv-builder)
   Compiling example-runner-wgpu v0.0.0 (/home/eddy/Projects/rust-gpu/examples/runners/wgpu)
    Finished `release` profile [optimized] target(s) in 3.58s
$ cargo build -p compiletests --release --no-default-features --features use-installed-tools
   Compiling rustc_codegen_spirv v0.9.0 (/home/eddy/Projects/rust-gpu/crates/rustc_codegen_spirv)
   Compiling compiletests v0.0.0 (/home/eddy/Projects/rust-gpu/tests)
    Finished `release` profile [optimized] target(s) in 2.64s
$ cargo build -p example-runner-wgpu --release --no-default-features --features use-installed-tools
   Compiling rustc_codegen_spirv v0.9.0 (/home/eddy/Projects/rust-gpu/crates/rustc_codegen_spirv)
   Compiling spirv-builder v0.9.0 (/home/eddy/Projects/rust-gpu/crates/spirv-builder)
   Compiling example-runner-wgpu v0.0.0 (/home/eddy/Projects/rust-gpu/examples/runners/wgpu)
    Finished `release` profile [optimized] target(s) in 3.44s
```

I did try re-adding the build-dep you removed (with version `2` instead of `1`), and nothing changed, so you didn't make anything worse (I was expecting this to have bitrotten anyway).